geopandas/tools/sjoin.py

```diff
@@ -186,6 +180,351 @@ def find_contains(a1, a2):
+
+        * 'left': use keys from left_df; retain only left_df geometry column
+        * 'right': use keys from right_df; retain only right_df geometry column
+        * 'inner': use intersection of keys from both dfs; retain only
+          left_df geometry column
+    lsuffix : string, default 'left'
+        Suffix to apply to overlapping column names (left GeoDataFrame).
+    rsuffix : string, default 'right'
+        Suffix to apply to overlapping column names (right GeoDataFrame).
+    max_search_neighbors : int, default None
+        Number of nearest neighbors to check for proximity.
+        Useful if you do not want to use search_radius.
+        note:: using a very small number (<5) can cause unexpected results
+          due to the implementation of the underlaying spatial index.
+          Geometries are converted to bounding boxes, thus results for the
+            nearest geometry may unexpected. Using a larger number mitigates
+            this, but using too large of a number will provide little performance
+            improvement.
+          If possible, you may want to run at least once with
+            max_search_neighbors=None to validate results for your data.
+    search_radius : int or float, default None
+        Restricts search to a certain radius.
+        This can significantly speed up execution.
+        If using a tight search radius, it is recommended that you do not use the
+          max_search_neighbors option as it will not improve performance.
+        If you are not using search_radius, consider using max_search_neighbors
+          to speed up execution.
+    nearest_distances: bool, default False
+        If True, report the distance for each match in new column
+          named "nearest_distances".
+    """
+    # ------------------------------ CHECK INPUTS ------------------------------
+    allowed_hows = ["left", "right", "inner"]
+    if how not in allowed_hows:
+        raise ValueError(
+            '`how` was "%s" but is expected to be in %s' % (how, allowed_hows)
+        )
+
+    _basic_checks(left_df, right_df)
+
+    # ----------------- RENAME INDEXES FOR RTREE COMPATIBILITY -----------------
+    (
+        left_df,
+        right_df,
+        left_index_name,
+        right_index_name,
+        index_left,
+        index_right,
+    ) = _rename_indexes(left_df, right_df, lsuffix, rsuffix)
+
+    # ------------------------ COMPUTE SPATIAL JOIN ----------------------------
+
+    # get spatial index
+    tree_idx = right_df.sindex
+
+    # validate max_search_neighbors and search_radius params
+    if max_search_neighbors is not None:
+        assert max_search_neighbors >= 1 and isinstance(
+            max_search_neighbors, int
+        ), "max_search_neighbors must be an integer and >= 1"
+        # warn about using rtree < 0.9.4 and max_search_neighbors option
+        # see https://github.com/Toblerity/rtree/pull/141
+
+        if [int(i) for i in RTREE_VERSION.split(".")] < [0, 9, 4]:
```

Comment:
In the rest of GeoPandas we do these check using `from distutils.version import LooseVersion`,(https://github.com/geopandas/geopandas/blob/master/geopandas/_compat.py) can you use that instead of this check?

Reply:
Will do